subsys/net/ip/net_if.c

```diff
@@ -4664,6 +4676,12 @@ static void notify_iface_up(struct net_if *iface)
 	    !(l2_flags_get(iface) & NET_L2_POINT_TO_POINT)) {
 		iface_ipv6_start(iface);
 		net_ipv4_autoconf_start(iface);
+
+		/* Make sure that we update the IPv6 addresses and join the
+		 * multicast groups.
+		 */
+		rejoin_multicast_groups(iface);
+		net_if_start_dad(iface);
```

Comment:
`iface_ipv6_start()` already covers DAD triggering (and checks for NET_IF_IPV6_NO_ND flag, which is important for OT case). I think it'd be enough to just move `rejoin_multicast_groups()` before we call that.

Reply:
Ok, I see. Makes sense.